nixos/modules/services/mail/postfix.nix

```diff
@@ -354,6 +328,18 @@ in
         ";
       };
 
+      config = mkOption {
+        type = with types; attrsOf (either bool (either str (listOf str)));
```

Comment:
This feel like a little wild type, `types.attrs` might be enough and could provide more appropriate declaration merging than `either`.

The approach used in the [influxdb module](https://github.com/NixOS/nixpkgs/blob/0906a0f19718605a8074c1c1bf0b3368f8e0d09c/nixos/modules/services/databases/influxdb.nix#L8-L97) for a similar setting, defining a configuration attribute set and update with an `attrs` option, might interest you as it can be a good trade-off of verbosity and complexity.
Or there is the possibility to use a `submodule` to enforce the type checking, but at the cost of verbosity.